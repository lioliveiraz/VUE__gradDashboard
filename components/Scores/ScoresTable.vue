<template>
  <table class="g-table-wrapper" data-testId="score_table">
    <thead>
      <tr class="font-bold">
        <th class="g-table-cell">N</th>

        <th class="g-table-cell">
          {{ $t("TABLE_HEAD_ASSCODE") }}
        </th>

        <th class="g-table-cell">
          {{ $t("TABLE_HEAD_SCORES") }}
        </th>
        <th class="g-table-cell">
          {{ $t("TABLE_HEAD_PASSED") }}
        </th>
      </tr>
    </thead>
    <tbody>
      <tr class="g-table-cell" v-for="({ code, score }, i) in scores" :key="i">
        <td class="g-table-cell" data-testId="score_table--index">
          {{ i + 1 }}
        </td>
        <td class="g-table-cell" data-testId="score_table--code">
          {{ code }}
        </td>
        <td class="g-table-cell" data-testId="score_table--score">
          {{ score }}
        </td>
        <td
          :class="
            score > 140
              ? ' g-table-cell  g-cell-passedAss'
              : ' g-table-cell  g-cell-failedAss'
          "
        >
          {{ score > 140 ? $t("YES") : $t("NO") }}
        </td>
      </tr>
    </tbody>
  </table>
</template>

<script>
export default {
  props: { scores: Array },
  layout: "dash_layout",
  middleware: "courses",

  created() {},
};
</script>

<style>
</style>