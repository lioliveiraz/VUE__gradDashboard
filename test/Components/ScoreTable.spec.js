import ScoresTable from '../../components/Scores/ScoresTable.vue';
import { shallowMount } from '@vue/test-utils';

describe("<ScoresTable>", () => {
    let scores, wrapper;
    beforeEach(async () => {
        scores = [{
            code: "847843",
            score: "3"
        }];
        wrapper = await shallowMount(ScoresTable, {
            propsData: { scores }, mocks: {
                $t: (msg) => msg
            }
        });
    });

    it('should render correctly', () => {
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('score_table')).toBeTruthy();
        expect(wrapper.find('score_table--code')).toBeTruthy();
        expect(wrapper.find('score_table--score')).toBeTruthy();

    });
    it('receive props correctly', () => {

        expect(wrapper.props()).toEqual({ scores });
        expect(wrapper.text()).toContain("847843");
    });
});