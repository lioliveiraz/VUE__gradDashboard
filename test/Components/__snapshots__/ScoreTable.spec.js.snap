// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ScoresTable> should render correctly 1`] = `undefined`;
